test/integration_test/fafb_matrix_test.go

```diff
@@ -0,0 +1,331 @@
+	//	// TODO: should have multiple valid FAs
+	//	Spec:     simpleClusterSpec,
+	//},
+	// C56413: https://portworx.testrail.net/index.php?/cases/view/56413
+	// Install PX through operator with both valid and invalid FlashArrays
+	//{
+	//	ID:                      "C56413",
+	//	ShouldStartSuccessfully: true,
+	//	// TODO: should have at least one valid FA and at least one invalid FA (invalid mgmt endpoint or API token)
+	//	Spec:     simpleClusterSpec,
+	//},
+}
+
+/*var flashArrayNegativeInstallTests = []TestrailCase{
+	// C54982: https://portworx.testrail.net/index.php?/cases/view/54982
+	// Install PX with invalid credentials
+	// C55004: https://portworx.testrail.net/index.php?/cases/view/55004
+	// Uninstall PX through operators with Pure arrays, after failed install
+	{
+		ID:                      "C54982-C55004",
+		ShouldStartSuccessfully: false,
+		// TODO: needs invalid credentials
+		Spec:     simpleClusterSpec,
+	},
+	// C54981: https://portworx.testrail.net/index.php?/cases/view/54981
+	// Install PX with only FlashBlades w/o iSCSI packages installed.
+	{
+		ID:                      "C54981",
+		ShouldStartSuccessfully: false, // assuming running with no other cloud provider creds
+		// TODO: needs some way to uninstall and reinstall iSCSI
+		// TODO: needs only FlashBlades
+		Spec:     simpleClusterSpec,
+	},
+	// C54983: https://portworx.testrail.net/index.php?/cases/view/54983
+	// Install PX on node with missing required utilities
+	{
+		ID:                      "C54983",
+		ShouldStartSuccessfully: false,
+		// TODO: needs some way to uninstall utilities or files and add them back afterwards
+		Spec:     simpleClusterSpec,
+	},
+}*/
+
+func TestFAFBPositiveInstallation(t *testing.T) {
+	for _, testCase := range flashArrayPositiveInstallTests {
+		t.Run(testCase.ID, matrixTest(testCase))
+	}
+}
+
+func matrixTest(c TestrailCase) func(*testing.T) {
+	return func(t *testing.T) {
+		// Check that we have enough backends to support this test: skip early
+		fleetBackends := GenerateFleetOrSkip(t, pxNamespace, c.BackendRequirements)
+
+		// Get versions from URL
+		logrus.Infof("Get component images from versions URL")
+		imageListMap, err := testutil.GetImagesFromVersionURL(pxSpecGenURL)
+		require.NoError(t, err)
+
+		// Construct Portworx StorageCluster object
+		cluster, err := constructStorageCluster(pxSpecGenURL, imageListMap)
+		require.NoError(t, err)
+
+		c.PopulateStorageCluster(cluster)
+
+		// Disable telemetry
+		if cluster.Spec.Monitoring == nil {
+			cluster.Spec.Monitoring = &v1.MonitoringSpec{}
+		}
+		cluster.Spec.Monitoring.Telemetry = &v1.TelemetrySpec{
+			Enabled: false,
+		}
+		// Add extra env variables
+		cluster.Spec.Env = append(cluster.Spec.Env, []k8sv1.EnvVar{
+			{
+				Name:  "DEFAULT_OSB_URL",
+				Value: "https://master-staging.portworx.dev/osb/billing/v1/register",
+			},
+			{
+				Name:  "PX_RELEASE_MANIFEST_URL",
+				Value: "https://install.portworx.com/2.8.0/version",
```

Comment:
should we use 2.9.0 here now?

Reply:
That's a good question. @piyush-nimbalkar a) should this value instead be taken from some argument to the tests, and b) is there any value that will always keep us up to date with the latest version, or will we have to keep changing the URL over time?

Reply:
In hindsight, there's an easily accessible `pxSpecGenURL` variable in the tests that gets passed in from the pod spec: I should probably just base this off of that. Changing...

Reply:
Agreed. It changes every release. One solution could be from an Env variable.